website/integrations: SeaTable: cleanup (#23338)
goauthentik/authentik: website/integrations/chat-communication-collaboration/seatable/index.md

The following placeholders are used in this guide:

- `seatable.company` is the FQDN of the self-hosted SeaTable installation.
- `authentik.company` is the FQDN of the authentik installation.

:::info SeaTable Cloud and self-hosted SeaTable
SeaTable supports SAML SSO for SeaTable Cloud Enterprise teams and self-hosted SeaTable Server Enterprise Edition installations. SeaTable Cloud requires domain verification in SeaTable Team Management, which is outside the scope of this guide.
:::

:::info
This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
:::

<Tabs
defaultValue="saas"
values={[
{ label: "SeaTable Cloud (SaaS)", value: "saas" },
{ label: "Self-hosted SeaTable", value: "self-hosted" },
]}>
<TabItem value="saas">

## authentik configuration

To support the integration of SeaTable Cloud with authentik, you need to create SAML property mappings and an application/provider pair in authentik.

> First, log in to SeaTable Cloud and navigate to **Team Management** > **Teams** > **Settings** > **Single Sign-On**. Keep the SeaTable-provided **Entity ID**, **Assertion Consumer Service (ACS) URL**, **Login (SSO URL)**, and **Logout** values available while creating the authentik provider.

Reviewer: These should probably be in preparation or a different section rather than under authentik configuration

> ### Create property mappings in authentik

Reviewer: ```suggestion
### Create property mappings
```

SeaTable Cloud requires SAML attributes named `contact_email`, `name`, and `uid`. Create three SAML provider property mappings for these attributes.

1. Log in to authentik as an administrator and open the authentik Admin interface.
2. Navigate to **Customization** > **Property Mappings** and click **Create**.
3. Select **SAML Provider Property Mapping** as the type and click **Next**.
4. Create a property mapping with the following values:
    - **Name**: `SeaTable contact_email`
    - **SAML Attribute Name**: `contact_email`
    - **Expression**:

        ```python
        return request.user.email
        ```

5. Click **Finish** to save the property mapping.
6. Repeat steps 2-5 to create the following additional property mappings:
    - **Name**: `SeaTable name`
    - **SAML Attribute Name**: `name`
    - **Expression**:

        ```python
        return request.user.name
        ```

    - **Name**: `SeaTable uid`
    - **SAML Attribute Name**: `uid`
    - **Expression**:

        ```python
        return request.user.uid
        ```

> ### Create an application and provider in authentik

Reviewer: ```suggestion
### Create an application and provider
```

<SAMLProvider20265Warning />

1. Log in to authentik as an administrator and open the authentik Admin interface.
2. Navigate to **Applications** > **Applications** and click **New Application** to open the application wizard.

>     - **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings. Note the application **Slug** because it will be required later.

Reviewer: ```suggestion
    - **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings. Note the application **Slug** because you will use it later as `<application_slug>`.
```

### Determine effective URL for metadata download

authentik's **metadata download URL** returns a 302 redirect, but SeaTable Cloud requires the effective URL. Run the following command to determine the effective URL:

```sh
curl -Ls -o /dev/null -w '%{url_effective}\n' "<metadata_download_URL>" 2>/dev/null
```

The output of this command will be required as the SeaTable metadata URL.

### Configure SeaTable Cloud

1. Log in to SeaTable Cloud and navigate to **Team Management** > **Teams** > **Settings** > **Single Sign-On**.
2. Configure the following settings:
    - **Metadata URL**: enter the effective metadata URL from authentik.
    - **Certificate**: upload or paste the signing certificate that you downloaded from authentik.
    - **Domain**: select the email domain that should use this SSO configuration.
3. Complete the DNS domain verification shown by SeaTable.

</TabItem>

<TabItem value="self-hosted">

## authentik configuration

To support the integration of self-hosted SeaTable with authentik, you need to create an application/provider pair in authentik.

> ### Create an application and provider in authentik

Reviewer: ```suggestion
### Create an application and provider
```

<SAMLProvider20265Warning />

1. Log in to authentik as an administrator and open the authentik Admin interface.
2. Navigate to **Applications** > **Applications** and click **New Application** to open the application wizard.

>     - **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings. Note the application **Slug** because it will be required later.

Reviewer: ```suggestion
    - **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings. Note the application **Slug** because you will use it later as `<application_slug>`.
```

Restart the SeaTable service or Docker container to apply the changes.

</TabItem>

</Tabs>

## Configuration verification

> To confirm that authentik is integrated correctly with SeaTable, log out of SeaTable and open the SeaTable integration from authentik. After authenticating, you should be redirected to SeaTable.

Reviewer: ```suggestion
To confirm that authentik is integrated correctly with SeaTable, log out of SeaTable and access SeaTable from the authentik application dashboard. You should be redirected to SeaTable.
```

Is this what you were trying to say or did I misunderstand?
